sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2150,7 +2150,8 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
 
   test("data source table created in InMemoryCatalog should be able to read/write") {
     withTable("tbl") {
-      sql("CREATE TABLE tbl(i INT, j STRING) USING parquet")
+      val provider = spark.sessionState.conf.defaultDataSourceName
```

Comment:
Hm .. how about just explicitly setting `spark.sql.sources.default` to `parquet` in all places rather than using the default? If it's set to, for example, `text`, this test becomes failed. I thought it's a bit odd a test it is dependent on a default value.

Reply:
This is `SQLQuerySuite`. The test case is correctly testing its **purpose**. Every data source have its own capability and limitation. Your example is only `text` data source's limitation supporting `a single column schema`, isn't it? For the other `csv/json/orc/parquet` will pass this specific test.

Reply:
So far, the purpose of this PR is **setting once in `SQLConf.scala`** to order to test a new data source to find out the limitation instead of **touching every data suite**.

BTW, `spark.sql.sources.default=parquet` doesn't help this existing code because the SQL has a fixed string `USING parquet`.